web/js/cartographedges.js

```diff
@@ -0,0 +1,127 @@
+var CG = CG || {};
+var storeddotslayer = [];
+var storedcurveslayer = [];
+
+ CG.showEdgesCityTooltip = function (mapX, mapY, properties) {
+        var id = properties.id;
+        var relatedPoints = '../point.json?id=' + id;
+
+        var randHue = 'rgb(' + (Math.floor(Math.random() * 256))
+                            + ',' + (Math.floor(Math.random() * 256))
+                            + ',' + (Math.floor(Math.random() * 256)) + ')';
+
+        var edgeHoverStyle =   {weight: 3,
+                                opacity: 0.95,
+                                smoothFactor: 1,
+                                attribution: 'edge'};
+        var edgeNeutralStyle = {color: randHue,
+                                weight: 1,
+                                opacity: 0.65,
+                                smoothFactor: 1,
+                                attribution: 'edge'};
+
+        $.getJSON(relatedPoints, function (data) {
+            CG.map.removeLayer(storeddotslayer);
+            CG.map.removeLayer(storedcurveslayer);
+
+            storeddots = [];
+            storedcurves = [];
+
+            var smallMarker = L.icon({
+            iconUrl: 'images/blackDot.png',
+            iconSize: [10,10]
+            });
+
+            coords = [];
+            names = [];
+
+            data[id].forEach(function (linkInfo) {
+                  var x = linkInfo.data.loc[0];
+                  var y = linkInfo.data.loc[1];
+                  marker = L.marker([x, y], {icon: smallMarker});
+                  storeddots.push(marker);
+                  coords.push([linkInfo.data.id, linkInfo.data.name, [x, y]]);
+            });
+            storeddotslayer = L.layerGroup(storeddots);
+            CG.map.addLayer(storeddotslayer);
+
+
+            var linkPairArray = [];
+            for(i=1; i < coords.length; i++) {
+                linkPairArray.push(coords[0]);
+                linkPairArray.push(coords[i]);
+            }
+
+
+            function drawCurves(){
```

Comment:
A few remaining requests:
- Please move drawCurves outside to an independent function outside of this so it is easier to read showEdgesCityTooltip
- Add some comments to make sure it's clear what's going on (see comment below).